src/main/java/seedu/address/logic/wanderlustlogic/wanderlustcommands/copy/CopyCommand.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.logic.wanderlustlogic.wanderlustcommands.copy;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.Command;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.CommandResult;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.exceptions.CommandException;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.travelplan.TravelPlan;
+import seedu.address.model.travelplanner.Model;
+
+
+public class CopyCommand extends Command {
+    public static final String COMMAND_WORD = "copy";
+
+    public static final String MESSAGE_USAGE =
+            "copy: Copies the activity identified by the index number used in the wish list to the travel plan "
+                    + "identified by the index number used in the travel planner.\n"
+                    + "Parameters: INDEX (must be a positive integer)\n";
+
+    public static final String MESSAGE_COPY_ACTIVITY_SUCCESS = "Copied activity %1$s to travel plan %1$s";
+    public static final String MESSAGE_NOT_WISHLIST = "Please goto wish list before copying activities";
+
+    private final Index activityIndex;
+    private final Index travelPlanIndex;
+
+    /**
+     * Constructor for MoveCommand.
+     *
+     * @param activityIndex index of activity to be copied.
+     * @param travelPlanIndex index of travel plan to add activity to.
+     */
+    public CopyCommand(Index targetIndex, Index travelPlanIndex) {
+        this.activityIndex = targetIndex;
+        this.travelPlanIndex = travelPlanIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        boolean isTravelPlan = model.isDirectoryTypeTravelPlan();
+        if (!isTravelPlan) {
+            List<Activity> filteredActivityList = model.getFilteredActivityList();
+            List<TravelPlan> travelPlanList = model.getFilteredTravelPlanList();
+
+            if (activityIndex.getZeroBased() >= filteredActivityList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_ACTIVITY_DISPLAYED_INDEX);
+            }
+
+            Activity activityToCopy = filteredActivityList.get(activityIndex.getZeroBased());
+            TravelPlan travelPlan = travelPlanList.get(travelPlanIndex.getZeroBased());
+
+            travelPlan.addTravelPlanObject(activityToCopy);
```

Comment:
Can this be implemented in model instead? Then call the model method instead of accessing travelPlan methods here directly. Treat model as a facade.